src/main/java/dev/esophose/playerparticles/particles/data/Vibration.java

```diff
@@ -2,14 +2,22 @@
 
 public class Vibration {
 
+    private org.bukkit.Vibration vibration;
     private final int duration;
 
+    public Vibration(org.bukkit.Vibration vibration) {
+        this.vibration = vibration;
+        this.duration = vibration.getArrivalTime();
+    }
     public Vibration(int duration) {
         this.duration = duration;
     }
 
     public int getDuration() {
-        return this.duration;
+        return duration;
```

Comment:
Please include `this.` in front of all instance variables for consistency with the rest of the project. This comment applies everywhere you made changes.